pkg/util/unique_name.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package util
+
+import "fmt"
+
+// GenerateUniqueConstraintName attempts to generate a unique constraint name
+// with the given prefix.
+// It will first try prefix by itself, then it will subsequently try
+// adding numeric digits at the end, starting from 1.
+func GenerateUniqueConstraintName(prefix string, nameExistsFunc func(name string) bool) string {
```

Comment:
This doesn't have anything to do with constraints. I also think it should live in the SQL package, not util, to make sure random other users don't start to have expectations about it and lock us into behaving the way we do right now.

Reply:
(doesn't have to be `pkg/sql` itself, but either that or a subpackage please!)

Reply:
maybe I should make it just generate unique name, because there are some uses in sqlbase. I could move it to there though, but it seems like a utility function.

Reply:
`util` is where things go to die. Make a new package with a better name, maybe!